TestNG Listener refactor

OliverHoenig opened the pull request:
# 📄 PR Description

<!--
Add a description for this pull request. Follow the checklist below.
- What is the purpose of the pull request?
- Illustrate the problem this PR solves and why this solution is appropriate.
  - Include an overview of the changes, ideally with 1-2 examples that illustrate the changes.
  - Include links to any additional context such as tracking issues or tickets.
-->
<!--- Describe your changes in detail including a relevant motivation and context for this change -->
This PR provides a fix for  #311.
Therefore, it implements a new TestNG Listener to change the status of a terminated test.
This change was needed because the status of a finished Test should never be changed. Doing this can introduce unwanted behavior with TestNG and other testing tools.


# ✒️ Checklist PR author

## Before submitting the PR, please make sure you do the following

### Prepare Code for PR
* [x] The automated tests were successfully executed locally (`mvn clean test`).
* [x] All commented-out code and debug statements have been removed.
* [x] A test is included that fails without this PR but passes with it.
* [x] The documentation in the code has been updated and corresponds to the team guidelines.
* [x] A person has been assigned to update the public documentation as soon as possible if this PR requires changes to it.
    * [x] [Qytera Github-Page (github.io)](https://github.com/Qytera-Gmbh/Qytera-Gmbh.github.io)

### Review Preparation
* [x] Review from appropriate team members is requested.

# 🔍 Checklist  PR reviewer

## Before accepting the PR, please make sure you do the following

### Check Preconditions
* [x] It is confirmed that the author has fulfilled all preconditions. (The review can only take place once this has been fulfilled.)

### Understand the Context
* [x] The PR description is read thoroughly to understand the changes.
* [x] It is verified that the changes fulfill the purpose described in the PR.
* [x] If the PR belongs to an issue or is intended to represent a user story, it is verified that the requirements and acceptance criteria are met.
- If things are unclear, please ask the author to add or adjust them in the description or as a comment. 

### Code Quality, Style & Functionality
* [x] It is ensured that the code adheres to the projects' [coding standards and style guides](https://qytera.atlassian.net/wiki/spaces/QO/pages/3800727592/QTAF+-+Coding+Standards).
* [x] The code is clean, readable and well-organized. 
* [x] The commits are named according to [the team conventions](https://qytera.atlassian.net/wiki/spaces/WIS/pages/3861348360/Commit+und+PR+Konventionen+f+r+QTAF). (If not, the author is informed so that he/she can improve for the next time. There are no changes to the commit history required.)
* [x] It is ensured that there are no commented-out code or debug statements left.
* [x] The changed or added code correctly handles potential edge cases and errors.

### Feedback and Communication
- Provide constructive feedback and suggestions. Be specific about what needs to be improved or changed.
  Use the [commenting feature](https://docs.github.com/de/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request) for PRs by GitHub.
  
### Approval
- Approve the PR if all checks are satisfied.

OliverHoenig committed:
fix(#311): add EventListener for TestNG Invocation

The status fa TestCase should not be changed
after termination. For this case TestNG provides
 an InvocationEventListener.

OliverHoenig committed:
docs(TestNG Listeners): add JavaDocs

csvtuda reviewed (changes requested):
Looks good to me. The only thing that's missing are at least two test cases for the new listener, where you check the overriden method's behaviour. But seeing that you've crossed out that line in the PR as well, I assume there's a reason? Please let me know what you think.

OliverHoenig committed:
test(Xray): ignore test

Some tests failed because they depended on a Jira project
that no longer exists. Since the feature that was covered
by the test is not actively supported and mocking the
response did not lead to meaningful tests, the tests
were deactivated.

OliverHoenig committed:
test(InvocationListener): add tests

Test for the method of the TestNGInvocationEventListener have been added.
Therefore, it was necessary to manipulate the log index
during the tests because, depending on the failed
steps, the method under test behaves differently

OliverHoenig commented:
> Looks good to me. The only thing that's missing are at least two test cases for the new listener, where you check the overriden method's behaviour. But seeing that you've crossed out that line in the PR as well, I assume there's a reason? Please let me know what you think.

I had assumed that this was implicitly covered by the existing tests. But yes, you are right. I have now added tests.
